src/peergos/shared/user/fs/FileWrapper.java

```diff
@@ -1000,6 +1000,27 @@ public static int getNumberOfChunks(long size) {
                 );
     }
 
+    @JsMethod
+    public CompletableFuture<Boolean> calculateThumbnail(NetworkAccess network, Crypto crypto) {
```

Comment:
calculate suggests to me that something is returned. Maybe "updateThumbnail"?

Reply:
updateThumbnail is already taken (used as part of updateProfile)

Reply:
That's annoying, maybe calculateAndUpdateThumbnail? Probably existing updateThumbnail should have been called setThumbnail

Reply:
done